go/vt/vtgate/vtgate.go

```diff
@@ -1074,3 +1074,49 @@ func annotateBoundShardQueriesAsUnfriendly(queries []*vtgatepb.BoundShardQuery)
 		queries[i].Query.Sql = sqlannotation.AnnotateIfDML(q.Query.Sql, nil)
 	}
 }
+
+// unambiguousKeyspaceBKSIQ is a helper function used in the
+// ExecuteBatchKeyspaceIds method to determine the "keyspace" label for the
+// stats reporting.
+// If all queries target the same keyspace, it returns that keyspace.
+// Otherwise it returns an empty string.
+func unambiguousKeyspaceBKSIQ(queries []*vtgatepb.BoundKeyspaceIdQuery) string {
+	if len(queries) == 0 {
```

Comment:
switch len(queries) {
case 0:
  return ""
case 1:
  return queries[0].Keyspace
default:
...
}

may look better and more compact?

Reply:
I changed it switch. Looks better :)

In general, I try to avoid switch with longer blocks per case because than we start to see too much indented code. And similar like error handling (https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow), we should avoid that.

But here it definitely makes sense :)